ballerina-tests/tests/http2_listener_stop_tests.bal

```diff
@@ -0,0 +1,125 @@
+import ballerina/http;
+import ballerina/lang.runtime;
+import ballerina/test;
+
+service class HelloService {
+    *http:Service;
+
+    resource function get hello() returns string {
+        return "Hello";
+    }
+}
+
+@test:Config {}
+public function testHttp2ListenerStop1() returns error? {
+    http:Listener serviceEP = check new (http2ListenerStopTest1);
+    check serviceEP.attach(new HelloService());
+    check serviceEP.'start();
+
+    http:Client client1 = check new ("localhost:" + http2ListenerStopTest1.toString());
+    string|error response = client1->/hello;
+    if response is string {
+        test:assertEquals(response, "Hello");
+    } else {
+        test:assertFail(msg = "Found unexpected error: " + response.message());
+    }
+
+    check serviceEP.immediateStop();
+    runtime:sleep(5);
+
+    response = client1->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
+    }
+
+    http:Client client2 = check new ("localhost:" + http2ListenerStopTest1.toString());
+    response = client2->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
+    }
+}
+
+@test:Config {}
+public function testHttp2ListenerStop2() returns error? {
+    http:Listener serviceEP = check new (http2ListenerStopTest1);
+    check serviceEP.attach(new HelloService());
+    check serviceEP.'start();
+
+    http:Client client1 = check new ("localhost:" + http2ListenerStopTest1.toString(), http2Settings = { http2PriorKnowledge: true });
+    string|error response = client1->/hello;
+    if response is string {
+        test:assertEquals(response, "Hello");
+    } else {
+        test:assertFail(msg = "Found unexpected error: " + response.message());
+    }
+
+    check serviceEP.immediateStop();
+    runtime:sleep(5);
+
+    response = client1->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
+    }
+
+    http:Client client2 = check new ("localhost:" + http2ListenerStopTest1.toString(), http2Settings = { http2PriorKnowledge: true });
+    response = client2->/hello;
+    if response is error {
+        test:assertEquals(response.message(), "Something wrong with the connection");
+    } else {
+        test:assertFail(msg = "Found unexpected output: " + response);
+    }
+}
+
+@test:Config {}
+public function testHttp2SecuredListenerStop() returns error? {
```

Comment:
Shall we add another test case with HTTPS + prior knowledge? It should go in the same path as in this test. So ideally it is enough to add only this. Just to be on the safe side and to cover all the possible scenarios?

Reply:
Sure, will add one 👍

Reply:
Fixed via [f0efef6](https://github.com/ballerina-platform/module-ballerina-http/pull/1344/commits/f0efef63950997879165bdc55d451b9c50209325)